larreco/HitFinder/HitFinderTools/PeakFitterGaussian_tool.cc

```diff
@@ -270,13 +268,222 @@ namespace reco_tool {
+
+            if (fOutputHistograms) {
+              fCandPeakPositionHist->Fill(peakMean, 1.);
+              fCandPeakWidHist->Fill(peakWidth, 1.);
+              fCandPeakAmpitudeHist->Fill(amplitude, 1.);
+            }
+
+            Gaus.SetParameter(parIdx, amplitude);
+            Gaus.SetParameter(1 + parIdx, peakMean);
+            Gaus.SetParameter(2 + parIdx, peakWidth);
+            Gaus.SetParLimits(parIdx, 0.1 * amplitude, fAmpRange * amplitude);
+            Gaus.SetParLimits(1 + parIdx, meanLowLim, meanHiLim);
+            Gaus.SetParLimits(2 + parIdx, std::max(fMinWidth, 0.1 * peakWidth), widthLim);
+
+            parIdx += 3;
+          }
+
+          int newFitResult{-1};
+          try {
+            newFitResult = fHistogram.Fit(&Gaus, "QNWB", "", 0., roiSize);
+          }
+          catch (...) {
+            mf::LogWarning("GausHitFinder") << "Fitter failed finding a hit";
+          }
+
+          // If the fit result is not zero there was an error
+          if (!newFitResult) {
+            // ##################################################
+            // ### Getting the fitted parameters from the fit ###
+            // ##################################################
+            float newChi2PerNDF = (Gaus.GetChisquare() / Gaus.GetNDF());
+
+            if (newChi2PerNDF * fRefitImprovement < chi2PerNDF || newChi2PerNDF < fRefitThreshold) {
+              peakParamsVec.clear();
+
+              chi2PerNDF = (Gaus.GetChisquare() / Gaus.GetNDF());
+              NDF = Gaus.GetNDF();
+
+              int parIdx{0};
+              for (size_t idx = 0; idx < nGausNew; idx++) {
+                PeakFitParams_t peakParams;
+
+                peakParams.peakAmplitude = Gaus.GetParameter(parIdx);
+                peakParams.peakAmplitudeError = Gaus.GetParError(parIdx);
+                peakParams.peakCenter = Gaus.GetParameter(parIdx + 1) + float(startTime);
+                peakParams.peakCenterError = Gaus.GetParError(parIdx + 1);
+                peakParams.peakSigma = Gaus.GetParameter(parIdx + 2);
+                peakParams.peakSigmaError = Gaus.GetParError(parIdx + 2);
+
+                peakParamsVec.emplace_back(peakParams);
+
+                parIdx += 3;
+              }
+            }
+          }
+        }
+      }
+
       if (fOutputHistograms) fFitBaselineHist->Fill(Gaus.GetParameter(3 * nGaus), 1.);
     }
-#if 0
-    Gaus.Delete();
-#endif // 0
     return;
   }
 
+  ICandidateHitFinder::HitCandidate PeakFitterGaussian::FindRefitCand(
+    const TF1& fittedGaus,
+    const std::vector<float>& waveform,
+    const int startTime,
+    const int roiSize,
+    const ICandidateHitFinder::HitCandidateVec& hitCandidateVec,
+    const PeakParamsVec& fittedPeakVec) const
+  {
+
+    // Find the candidate that was shifted most by the fit
+    std::pair<ICandidateHitFinder::HitCandidate, PeakFitParams_t> shiftedHitCanPair =
+      FindShiftedGaussian(hitCandidateVec, fittedPeakVec);
```

Comment:
Suggest a structured binding for easier use:

```suggestion
    auto const [candidate, params] = FindShiftedGaussian(hitCandidateVec, fittedPeakVec);
```

and then replace `shiftedHitCanPair.first` with `candidate` and `shiftedHitCanPair.second` with `params`.